lib/pure/asynchttpserver.nim

```diff
@@ -65,14 +68,55 @@
 ##        bodyLength = 0
 ##      if req.reqMethod == HttpPost:
 ##        contentLength = req.headers["Content-length"].parseInt
-##        if contentLength < 8*1024: # the default chunkSize
-##          # read the request body at once
-##          let body = await req.bodyStream.readAll();
-##          bodyLength = body.len
-##        else:
-##          # read 8*1024 bytes at a time
-##          while (let data = await req.bodyStream.read(); data[0]):
-##            bodyLength += data[1].len
+##        # Read 8*1024 bytes at a time
+##        for length, data in req.bodyStream():
+##          let content = await data
+##          if length == content.len:
+##            bodyLength += content.len
+##          else:
+##            # Handle exception
+##            await req.respond(Http400,
+##              "Bad Request. Data read has a different length than the expected.")
+##            return
+##      await req.respond(Http200, htmlpage(contentLength, bodyLength))
+##
+##    let server = newAsyncHttpServer(maxBody = 10485760, stream = true) # 10 MB
+##    waitFor server.serve(Port(8080), cb)
+##
+## With default option stream (false is the default):
+## --------------------------------------------------
+##
+## .. code-block::nim
+##    import asynchttpserver, asyncdispatch
+##    import strutils, strformat
+##
+##    proc htmlpage(contentLength, bodyLength: int): string =
+##      return &"""
+##    <!Doctype html>
+##    <html lang="en">
```

Comment:
Excessive example, can we shorten it somehow?

Reply:
I already removed one of the examples.